e2e/ansible/playbooks/resiliency/test-k8s-master-failure/k8s-master-failure.yml

```diff
@@ -0,0 +1,311 @@
+
+
+     - name: 4b) Copy job configuration files to jenkins-pod from master.
+       shell: >
+         source ~/.profile; kubectl cp {{result_kube_home.stdout}}/{{item}} 
+         {{namespace}}/{{ jenkins_pod_name.stdout }}:/var/jenkins_home/
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       with_items: "{{ job_config_files }}"
+
+
+
+     - name: 5) Run the config_change.sh file for required changes inside pod.
+       shell: >
+          source ~/.profile; kubectl exec {{  jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "sh /var/jenkins_home/config_change.sh"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+
+     - name: Retrieve Password of jenkins running pods.
+       shell: >
+          source ~/.profile; kubectl exec  {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "cat /var/jenkins_home/secrets/initialAdminPassword"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: jenkins_password
+
+
+     - name: 5a) Restart the jenkins pod for running jenkins actions.
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 restart
+          --username admin --password {{ jenkins_password.stdout }}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+ 
+     - name: wait for 60 seconds.
+       wait_for:
+          timeout: 60
+
+
+     - name: 6)  Create and Build one sample job (Hello World)  inside pod.
+       shell: >
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s
+         http://localhost:8080 create-job sample < /var/jenkins_home/template.xml";
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace  }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 build sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+
+     - name: 6a)  Running a build job. 
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 console sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: First_Build
+
+
+     - name: Bring down the kubemaster.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+       vars:
+         vagrant_op: halt --force
+         nodename: kubemaster-01
```

Comment:
Lets not use vagrant to bring down the host.

Reply:
we are including test into run-test.yml ...to skip any test which is having vagrant operations.